Allowed configurable urls for frontend apps

rshbhgrg opened the pull request:
- current config for frontend apps doesn't allow easy switching of CDN for urls used to load them
- this change allows pro to switch out the CDN easily, and also avoids double use of version value
- also extends config to include url needed for loading CSS for frontend app

daniellockyer commented on ghost/core/core/shared/config/defaults.json:
```diff
@@ -137,7 +137,8 @@
-        "url": "https://unpkg.com/@tryghost/sodo-search@~1.0.0/umd/sodo-search.min.js",
+        "url": "https://unpkg.com/@tryghost/sodo-search@~{version}/umd/sodo-search.min.js",
+        "css": "https://unpkg.com/@tryghost/sodo-search@~{version}/umd/main.css",
         "version": "1.0.0"
```
Should we start truncating this to `major.minor` like the Portal one? It'll still resolve to the latest in unpkg and it should also work in jsdelivr because the expanded URL becomes `https://cdn.jsdelivr.net/npm/@tryghost/sodo-search@1.0/umd/sodo-search.min.js`

rshbhgrg committed:
ℹ️  Updated default config and CDN for frontend apps

refs https://github.com/TryGhost/Team/issues/1719

- switches default CDN from unpkg to jsDelivr for better reliability
- current config for frontend apps doesn't allow easy switching of CDN for frontend urls
- allows easy switch of base CDN for frontend apps via config
- fixes double use of version string in the config
- extends config to include url needed for loading styles for frontend app instead of hardcoded urs

rshbhgrg committed:
Updated frontend apps to use new config

refs https://github.com/TryGhost/Team/issues/1719

- creates scripts injected for portal/search/comments using new config format
- adds `data-styles` to search/comments script for adding external styles URL

daniellockyer commented on ghost/core/core/frontend/helpers/ghost_head.js:
```diff
@@ -47,9 +48,24 @@ function getMembersHelper(data, frontendKey) {
+    }
+    const dataAttributes = getDataAttributes(attributes);
+
+    if (colorString) {
```
Was this leftover from something? `dataAttributes` is a string at this point 🤔

rishabhgrg replied:
Ugh yea thanks, it was exactly that, was supposed to be deleted! Fixed

daniellockyer commented on ghost/core/core/frontend/helpers/comments.js:
```diff
@@ -38,18 +39,20 @@ async function comments(options) {
+        'app-version': appVersion,
         'comments-enabled': commentsEnabled
     };
 
```
Can we switch to the new util `getDataAttributes` util below? (sorry, GitHub won't let me comment on those lines 🙃)

rishabhgrg replied:
Done ✅

rshbhgrg committed:
Updated stylesheet url handling for frontend apps

refs https://github.com/TryGhost/Team/issues/1719

- updates search and comments app to use configurable stylesheet url instead of hardcoded value
